clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/producer/ProducerCompressionTest.java

```diff
@@ -0,0 +1,185 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.producer;
+
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.test.ClusterInstance;
+import org.apache.kafka.common.test.api.ClusterConfigProperty;
+import org.apache.kafka.common.test.api.ClusterTest;
+import org.apache.kafka.common.test.api.ClusterTestDefaults;
+import org.apache.kafka.common.test.api.Type;
+import org.apache.kafka.server.config.ServerLogConfigs;
+import org.apache.kafka.test.TestUtils;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static kafka.utils.TestUtils.consumeRecords;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+
+@ClusterTestDefaults(
+        types = {Type.KRAFT},
+        serverProperties = {
+            @ClusterConfigProperty(key = ServerLogConfigs.AUTO_CREATE_TOPICS_ENABLE_CONFIG, value = "false")
+        }
+)
+class ProducerCompressionTest {
+
+    private final String topicName = "topic";
+    private final int numRecords = 2000;
+
+    /**
+     * testCompression
+     *
+     * Compressed messages should be able to sent and consumed correctly
+     */
+    @ClusterTest
+    void testCompression(ClusterInstance cluster) {
+        Set<String> compressionSet = Set.of("none", "gzip", "snappy", "lz4", "zstd");
+        Set<String> protocolSet = Set.of("consumer", "classic");
+
+        compressionSet.forEach(compression -> {
+            protocolSet.forEach(protocol -> {
+                try {
+                    processCompressionTest(cluster, compression, protocol);
+                } catch (InterruptedException | ExecutionException e) {
+                    fail(e);
+                }
+            });
+        });
+    }
+
+
+    void processCompressionTest(ClusterInstance cluster, String compression, String protocol) throws InterruptedException,
+            ExecutionException {
+        String compressionTopic = topicName + "_" + protocol + "_" + compression;
```

Comment:
It's better to insert once and use different consumer protocol to validate it, so we don't need to insert 10 times (5 compression * 2 protocols). W can just insert 5 times (5 compression).